api/src/certification/configuration/infrastructure/repositories/framework-challenges-repository.js

```diff
@@ -0,0 +1,56 @@
+import { DomainTransaction } from '../../../../shared/domain/DomainTransaction.js';
+import { NotFoundError } from '../../../../shared/domain/errors.js';
+import { CertificationFrameworksChallenge } from '../../domain/models/CertificationFrameworksChallenge.js';
+import { FrameworkChallenges } from '../../domain/models/FrameworkChallenges.js';
+
+/**
+ * @param {Object} params
+ * @param {number} params.versionId
+ * @returns {Promise<FrameworkChallenges>}
+ * @throws {NotFoundError}
+ */
+export async function getByVersionId({ versionId }) {
+  const knexConn = DomainTransaction.getConnection();
+
+  const certificationFrameworksChallengesDTO = await knexConn('certification-frameworks-challenges')
+    .select('discriminant', 'difficulty', 'challengeId', 'versionId')
+    .where({ versionId })
+    .orderBy('challengeId');
+
+  if (certificationFrameworksChallengesDTO.length === 0) {
+    throw new NotFoundError('Framework challenges do not exist for this version');
+  }
+
+  return _toDomain({ certificationFrameworksChallengesDTO });
+}
+
+/**
+ * @param {FrameworkChallenges} frameworkChallenges
+ * @returns {Promise<void>}
+ */
+export async function save(frameworkChallenges) {
```

Comment:
Par coherence avec les autres repository, plutot utiliser le mot "update"

```suggestion
export async function update(frameworkChallenges) {
```

Reply:
Done ✅